resolver/resolver.go

```diff
@@ -29,9 +29,7 @@ type Resolver struct {
 	rs      *records.RecordGenerator
 	rsLock  sync.RWMutex
 	rng     *rand.Rand
-
-	// pluggable external DNS resolution, mainly for unit testing
-	extResolver exchanger.Exchanger
+	exchanger.Forwarder
```

Comment:
Yeah, in this case that's fine since it's becoming privatised. But when your API consists of anonymous functions being passed around, it's useful to have a more descriptive utility method attached to it so that you can name the functions other things than verbs.

If you look closer, there is a language mismatch. `NewForwarder` returns a `Forwarder` which is stored in a field with a verb name `forward`. With the utility method, this goes away.